superset-frontend/plugins/plugin-chart-pivot-table/src/plugin/buildQuery.ts

```diff
@@ -30,7 +30,10 @@ import {
 import { PivotTableQueryFormData } from '../types';
 
 export default function buildQuery(formData: PivotTableQueryFormData) {
-  const { groupbyColumns = [], groupbyRows = [] } = formData;
+  const { groupbyColumns = [], groupbyRows = [], extra_form_data } = formData;
```

Comment:
can we write a small unit test to for buildQuery to verify the grains are being properly set?

Reply:
Thank you for the references @hughhhh. I added the tests with modifications. 

One tip: you can ask Chat GPT to avoid using `describe` and `it` blocks to generate tests that conform to our [best practices](https://github.com/apache/superset/wiki/Testing-Guidelines-and-Best-Practices#avoid-nesting-when-youre-testing) 😉